com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java

```diff
@@ -244,12 +238,23 @@ private boolean shouldUpdateTags(CloudServiceExtended service, List<String> exis
         if (service.isUserProvided()) {
             return false;
         }
-        existingServiceTags = ObjectUtils.defaultIfNull(existingServiceTags, Collections.<String> emptyList());
+        existingServiceTags = ObjectUtils.defaultIfNull(existingServiceTags, Collections.<String>emptyList());
         return !existingServiceTags.equals(service.getTags());
     }
 
-    private boolean shouldUpdateCredentials(CloudServiceExtended service, Map<String, Object> credentials) {
-        return !Objects.equals(service.getCredentials(), credentials);
+    private boolean shouldUpdateCredentials(CloudServiceExtended service, CloudService existingService, CloudControllerClient client) {
+        if (existingService == null) {
```

Comment:
The existing service will be never null here. Check the if statement where this method is invoked.